p2p/protocol/identify/id.go

```diff
@@ -31,9 +33,27 @@ const ID = "/ipfs/id/1.0.0"
 
 // LibP2PVersion holds the current protocol version for a client running this code
 // TODO(jbenet): fix the versioning mess.
+// XXX: Don't change this till 2020. You'll break all go-ipfs versions prior to
```

Comment:
We should make this configurable too. Many projects using libp2p don't care about compatibility with IPFS and their networks are deliberately segregated.

Reply:
See #714